misc/launchd/org.nixos.nix-daemon.plist.in

```diff
@@ -14,7 +14,12 @@
     <key>RunAtLoad</key>
     <true/>
     <key>Program</key>
-    <string>@bindir@/nix-daemon</string>
+    <string>/bin/sh</string>
+    <key>ProgramArguments</key>
+    <array>
+      <string>-c</string>
+      <string>/bin/wait4path @bindir@/nix-daemon &amp;&amp; @bindir@/nix-daemon</string>
+    </array>
```

Comment:
The way I understand the `ProgramArguments` key in launchd plist files, the binary needs to be the first argument, because `ProgramArguments` is directly passed as `argv` to `execv`.

So I think this should read:
```
<key>ProgramArguments</key>
<array>
    <string>/bin/sh</string>
    <string>-c</string>
    <string>/bin/wait4path @bindir@/nix-daemon && exec @bindir@/nix-daemon</string>
</array>
```